exchanges/graphcache/src/operations/shared.ts

```diff
@@ -290,6 +296,19 @@ export class SelectionIterator {
   }
 }
 
+const isSuperType = (
```

Comment:
How would this work for an abstract type which is extending an abstract type i.e. a union of interfaces etc. basically this might need to be recursive

Reply:
@JoviDeCroock I think we can only have union of concrete object types
https://spec.graphql.org/draft/#sec-Unions.Type-Validation

And if interface A implements interface B, then an object C implements A must also declare that it implements B.
https://spec.graphql.org/draft/#sec-Objects.Type-Validation

From this, I believe we don't need recursively check for super type here (possibleTypes from introspection should contains all subtypes)